source/agora/test/Base.d

```diff
@@ -1101,9 +1101,11 @@ public class TestAPIManager
                     (file, line, height, idx,
                         this.clients[idx].getBlockHeight())));
 
-        retryFor(client_idxs.map!(idx =>
-            this.clients[idx].getAllBlocks()).uniq().count() == 1,
-            5.seconds,
+        Block[] all_blocks = cast(Block[])(client_idxs.map!(idx =>
+            this.clients[idx].getAllBlocks()).joiner.array);
+        all_blocks.sort!((a, b) => a.header.hashFull > b.header.hashFull);
```

Comment:
we could get rid of the sorting/hashing and use some linear algorithm, if we want
(x is client_idxs.map!(idx => this.clients[idx].getAllBlocks())

```D
    int[] a = [1, 2, 3, 4];
    int[] b = [10, 20, 30, 40];
    int[] c = [100, 200, 300, 400];
    auto x = [a, b, c];
    iota(x.length).map!(i => x.transversal(i)).joiner.writeln;
```